### PR TITLE
Replace reactable with react-table

### PR DESCRIPTION
**Pull request by seaninryan**
I updated `ImageListView` to use [react-table](https://github.com/tannerlinsley/react-table). 

I pretty much just did the minimum to get it working with sorting on, etc. but just uses the default `react-table` styles etc.

I'm going to try and see what I can do w.r.t. filtering

If you guys like it I can apply it to all of the other pages too

### CODE REVIEW

**Comment by ehazlett** [comment]
Nice!  This is looking good!

**Comment by seaninryan** [comment]
Rebased from `origin/swarm-mode` and got Tom's image removal stuff to work with `react-table`

**Comment by tombee** [comment]
This is awesome..

Do you know if it's possible to make the table not show empty rows?

e.g.
2017-02-02-18-05-21_1128x553

**Comment by tombee** [comment]
Was just wondering if there was a way to not show empty rows.. but looking at it, there seems to be an option to do it.. cool :)

```
minRows: 0, // Ensure this many rows are always rendered, regardless of rows on page
```

It's nice that we can set our own defaults too.. according to the docs:

```javascript
import { ReactTableDefaults } from 'react-table'

Object.assign(ReactTableDefaults, {
  defaultPageSize: 10,
  minRows: 3,
  // etc...
})
```

**Comment by seaninryan** [comment]
Hmm @tombee just looking at the [`react-table` page](https://github.com/tannerlinsley/react-table)... looks like they're removing support for `minRows`

File: ui/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "react-transform-catch-errors": "^1.0.2",
     "react-transform-hmr": "^1.0.4",
     "reactable": "^0.14.1",
+    "react-table": "^3.1.0",
```
Comment (tombee) [inline ui/package.json]: Apparently `5.0.0` is the latest release according to: https://www.npmjs.com/package/react-table, think we should upgrade to that?

**Review by tombee — APPROVED** [review]

File: ui/package.json (hunk 1 above)
Comment (seaninryan) [reply ui/package.json]: Ah. I thought 3.1.0 was the latest based off the RELEASES tab on their GitHub project

Can try it out l8r

**Comment by seaninryan** [comment]
- applied `react-table`/`ReactTable` to all table elements
- removed references to `reactable`
- updated to version 5.0.0 of `react-table`
- I disabled table filtering on all of these pages for now. Will come back to that problem when I get this PR out of the way

Couple of problems that I think I've introduced :confused: :
- clicking through to the `inspect/` pages fails with an error like:
```
Uncaught TypeError: Cannot read property 'id' of undefined
    at ImageInspectView.componentDidMount (ImageInspectView.js:19)
```
- after adding a `service` end up with a blank `ContentPane`
- wasn't able to test out `NodeInspectView.js`, `RegistryInspectView.js`, ServiceInspectView.js` (wasn't able to navigate to those pages). Presume that these won't work as things stand.

File: ui/src/components/accounts/AccountListView.js [hunk 2]
```diff
@@ -70,22 +70,22 @@ class AccountListView extends React.Component {
...
-        <Td column="Username" value={account.username}>
-          <Link to={`/accounts/inspect/${account.username}`}>{account.username}</Link>
-        </Td>
-        <Td column="First Name">{account.first_name}</Td>
-        <Td column="Last Name">{account.last_name}</Td>
-        <Td column="Roles">{account.roles.join(', ')}</Td>
-      </Tr>
-    );
-  };
+//  renderAccount = (account) => {
```
Comment (tombee) [inline ui/src/components/accounts/AccountListView.js:73]: Can probably just remove this now if it's no longer required? :)

**Comment by tombee** [comment]
@seaninryan 
I don't think you introduced the inspect page issues, I think that was me when I upgraded react-router (`this.props.params.id` -> `this.props.match.params`).  I believe these issues should be fixed in the PR I have open (e.g. https://github.com/shipyard/shipyard/pull/958/files#diff-c9a1ead50e81e6362c24a83a9296307dR18).

I think this is good to go as-is.

**Comment by tombee** [comment]
LGTM

**Review by tombee — APPROVED** [review]

**Pull request merged**